src/actions/commands/actions.ts

```diff
@@ -3270,6 +3275,22 @@ class ActionJoinVisualMode extends BaseCommand {
   }
 }
 
+@RegisterAction
+class ActionJoinVisualBlockMode extends BaseCommand {
+  modes = [Mode.VisualBlock];
+  keys = ['J'];
+  mightChangeDocument = true;
+
+  public async exec(position: Position, vimState: VimState): Promise<VimState> {
+    const start = vimState.cursorStartPosition;
+    const end = vimState.cursorStopPosition;
+
+    vimState.currentRegisterMode = RegisterMode.CharacterWise;
+    vimState = await new ActionJoin().execJoinLines(start, end, vimState, 1);
```

Comment:
Could you condense this like so: `return new ActionJoin().execJoinLines(vimState.cursorStartPosition, vimState.cursorStopPosition, vimState, 1);`?

Sorry to nitpick again; really appreciate the contributions! 🙂

Reply:
No problem, I think this is good, thank you 😀